src/utils/SocketClient.hxx

```diff
@@ -46,6 +46,19 @@
 #include "utils/Atomic.hxx"
 #include "utils/format_utils.hxx"
 
+/** Connection status that can be sent back to the "owner" of the socket so
+ * it can update display or status information while the connection attempts
+ * are progressing.
+ */
+enum class SocketStatus
```

Comment:
please move this inside SocketClient and call it Status.
(The reason is that we are in the global namespace and we should be careful about namespace pollution)